API/Controllers/UserController.cs

```diff
@@ -166,5 +172,44 @@ public async Task<IActionResult> DeleteAccount(int userId)
             userService.Save();
             return Ok();
         }
+
+        /// <summary>
+        ///     Delete all user data
+        /// </summary>
+        /// <returns></returns>
+        [HttpDelete("{userId}/allData")]
+        [Authorize(Policy = nameof(Defaults.Scopes.UserWrite))]
+        public async Task<IActionResult> DeleteAllUserData(int userId)
+        {
+            if(await userService.FindAsync(userId) == null)
+            {
+                ProblemDetails problem = new ProblemDetails
+                {
+                    Title = "Failed getting the user account.",
+                    Detail = "The database does not contain a user with this student id.",
+                    Instance = "TODO-CHANGE-TO-GENERATED-INSTANCE-CODE"
+                };
+                return NotFound(problem);
+            }
+            List<Project> projects = (List<Project>) await projectService.GetAllWithUsersAsync();
+            projects.ForEach(delegate(Project project) {
+                if(project.UserId.Equals(userId))
+                {
+                    project.UserId = -1;
```

Comment:
Unfortunately I don't think this will work very well, @wotwot563 do you have suggestions on how to handle GDPR requests. We want to keep the projects, while removing the user link.

This will break the get project behavior I think when retrieving a project that has no user associated with it.

Reply:
allow the project user to be nullabe and then set the user to null